File: languages/perl/examples/basic.pl
use strict;
use warnings;
use FindBin qw($Bin);
use lib "$Bin/../src";
use PositionTape qw(Generate GenerateMarkerComplete Validate);

my $exact = Generate(100);
my $marker_complete = GenerateMarkerComplete(1000);
my $validation = Validate($exact, 100);

print "$exact\n";
print length($marker_complete) . "\n";
print ($validation->{is_valid} ? "true" : "false") . "\n";

File: languages/python/examples/basic.py
from position_tape import Generate, GenerateMarkerComplete, Validate


def main() -> None:
    exact = Generate(100)
    marker_complete = GenerateMarkerComplete(1000)
    validation = Validate(exact, 100)

    print(exact)
    print(len(marker_complete))
    print(validation.is_valid)


if __name__ == "__main__":
    main()
